src/node/node_state.h

```diff
@@ -898,7 +901,10 @@ namespace ccf
             // Inform snapshotter of all signature entries so that this node can
             // continue generating snapshots at the correct interval once the
             // recovery is complete
-            snapshotter->record_committable(last_recovered_idx);
+            if (snapshotter->record_committable(last_recovered_idx))
+            {
```

Comment:
This isn't necessary here and will be remove altogether once https://github.com/microsoft/CCF/issues/3613 is implemented. The `GenesisGenerator` on line `857` may be removable too as unused?